pytensor/link/numba/dispatch/slinalg.py

```diff
@@ -311,3 +313,19 @@ def cho_solve(c, b):
         )
 
     return cho_solve
+
+
+@numba_funcify.register(BandedDot)
+def numba_funcify_BandedDot(op, node, **kwargs):
+    kl = op.lower_diags
+    ku = op.upper_diags
+    dtype = node.inputs[0].dtype
+
+    if dtype in complex_dtypes:
+        raise NotImplementedError(_COMPLEX_DTYPE_NOT_SUPPORTED_MSG.format(op=op))
+
+    @numba_njit
```

Comment:
I think this is what we should be doing instead/in addition of filtering the warnings. I just don't know what happens if a function that calls this (like Blockwise) and is itself trying to cache will do

```suggestion
    @numba_njit(cache=False)
```